### PR TITLE
terra home work #1

### PR DESCRIPTION
**Pull request by atonkyi**

### CODE REVIEW

File: my_instance.tf [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    user        = "ubuntu"
+    private_key = file("terra-key")
+    host        = self.public_ip
+  }
+
+}
+
+
+output "public_ipv4" {
+  value = aws_instance.intro
```
Comment (koshmanov) [inline my_instance.tf]: This part won't work. Terraform will throw an error.

File: terra-key [hunk 2]
```diff
@@ -0,0 +1,38 @@
+-----BEGIN OPENSSH PRIVATE KEY-----
```
Comment (koshmanov) [inline terra-key:1]: Try to keep private key file only on local machine. Don't add it into the repo.

File: my_instance.tf [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  instance_type          = "t2.micro"
+  availability_zone      = "us-east-1a"
+  key_name               = aws_key_pair.terra-key.key_name
+  vpc_security_group_ids = [aws_security_group.ec2_sg.name]
+  tags = {
+    Name    = "Terra-instance"
+    Project = "Terra"
+  }
+
+  provisioner "file" {
```
Comment (koshmanov) [inline my_instance.tf]: This is a good approach for docker installation 👍. 
Do you know another approach?

**Review by koshmanov — APPROVED** [review]
Good job 🔥

File: terra-key (hunk 2 above)
Comment (atonkyi) [reply terra-key:1]: undestood

File: my_instance.tf (hunk 3 above)
Comment (atonkyi) [reply my_instance.tf]: No, I've just found instruction from official site ... 
I didn't have to look for another way.

Maybe you kmow better approach ?

File: my_instance.tf (hunk 1 above)
Comment (atonkyi) [reply my_instance.tf]: Hmm ... will check

Comment (atonkyi) [reply my_instance.tf]: Fixed issue

File: my_instance.tf [hunk 4]
```diff
@@ -1,95 +1,28 @@
...
-}
-
-*/
-
-
-
-
 #Create key-pair for Ec2 instance
-/*resource "aws_key_pair" "terra-key" {
+resource "aws_key_pair" "terra-key" {
```
Comment (koshmanov) [inline my_instance.tf:2]: Don't use "-" in block's name. Use "_" instead.

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,22 @@
+variable "REGION" {
```
Comment (koshmanov) [inline variables.tf:1]: Variables names should be lowercase.

File: my_instance.tf [hunk 6]
```diff
@@ -122,34 +55,29 @@ resource "aws_secretsmanager_secret_version" "service_password" {
...
   ami                    = "ami-053b0d53c279acc90"
   instance_type          = "t2.micro"
   availability_zone      = "us-east-1a"
   key_name               = aws_key_pair.terra-key.key_name
   vpc_security_group_ids = [aws_security_group.ec2_sg.name]
-  tags = {
-    Name    = "Terra-instance"
-    Project = "Terra"
-  }*/
+  tags                   = merge(var.common_tags,{Name = "Ec2_instance"})
```
Comment (koshmanov) [inline my_instance.tf:71]: 👍

File: my_instance.tf [hunk 7]
```diff
@@ -122,34 +55,29 @@ resource "aws_secretsmanager_secret_version" "service_password" {
...
 
-}*/
+
+}
 
 
 #Create Ec2 instance
-/*resource "aws_instance" "intro" {
+resource "aws_instance" "intro" {
   ami                    = "ami-053b0d53c279acc90"
```
Comment (koshmanov) [inline my_instance.tf:66]: Do you know how to find AMI ID via AMI name?

File: my_instance.tf [hunk 8]
```diff
@@ -1,95 +1,28 @@
...
+}
 
 
 #Create security group for Ec2 instance
-/*resource "aws_security_group" "ec2_sg" {
+resource "aws_security_group" "ec2_sg" {
   name        = "ec2 sg"
   description = "allow ports: 22, 80, 443"
 
+  dynamic "ingress" {
```
Comment (koshmanov) [inline my_instance.tf:13]: 👍

**Review by koshmanov — APPROVED** [review]
Good job 👍

**Pull request merged**